src/app/components/HoldConfirmation/HoldConfirmation.jsx

```diff
@@ -32,6 +32,11 @@ class HoldConfirmation extends React.Component {
    */
   requireUser() {
     if (this.state.patron && this.state.patron.id) {
+      const queryID = this.props.location.query ? this.props.location.query.id : '';
```

Comment:
OK. My bad. I tested again and it actually works well!!!
it just will flash the confirmation page a bit and redirects you. I feel it should be done by the server and database to check if the jobID and your patron cookie somehow matches. Is this more ideal implement?

But we can use this as a client side fallback, and it actually works.